sql/connect/client/jdbc/src/main/scala/org/apache/spark/sql/connect/client/jdbc/SparkConnectStatement.scala

```diff
@@ -86,7 +88,10 @@ class SparkConnectStatement(conn: SparkConnectConnection) extends Statement {
     operationId = null
     resultSet = null
 
-    val df = conn.spark.sql(sql)
+    var df = conn.spark.sql(sql)
+    if (maxRows > 0) {
```

Comment:
0 is also meaningful user request, isn't it?

```scala
- if (maxRows > 0) {
+ if (maxRows >= 0) {
```

Reply:
according to the JDBC API javadoc (I pasted it in PR description)

> zero means there is no limit

Reply:
Oh, I missed that. Got it!